src/vmath.nim

```
@@ -327,51 +329,53 @@ proc angleBetween*(a, b: Vec3): float32 =
   arccos(dot)
 
 proc `[]`*(a: Vec3, i: int): float32 =
-  assert(i == 0 or i == 1 or i == 2)
+  if i > 2:
+    raise newException(IndexDefect, "Vec3 index out of bounds")
   if i == 0:
-    return a.x
+    a.x
   elif i == 1:
-    return a.y
-  elif i == 2:
-    return a.z
+    a.y
+  else:
+    a.z
 
 proc `[]=`*(a: var Vec3, i: int, b: float32) =
-  assert(i == 0 or i == 1 or i == 2)
+  if i > 2:
+    raise newException(IndexDefect, "Vec3 index out of bounds")
   if i == 0:
     a.x = b
   elif i == 1:
     a.y = b
-  elif i == 2:
+  else:
```

I think the extra else branch should just raise an error.